### PR TITLE
AsyncModbusTcp/UdpClient honors delay_ms == 0 (#1203)

### PR DESCRIPTION
**Pull request by Cavaler**

### CODE REVIEW

File: pymodbus/client/udp.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def protocol_lost_connection(self, protocol):
 
             self.connected = False
             self.protocol = None
-            if self.params.host:
```
Comment (Cavaler) [inline pymodbus/client/udp.py:161]: I don't have any UDP server, so haven't tried it

Comment (janiversen) [reply pymodbus/client/udp.py:161]: I think it is covered by our test suite.

Comment (janiversen) [reply pymodbus/client/udp.py:161]: and just for information, pymodbus have an excellent server, which are very good for testing cases like this.

**Review by janiversen — APPROVED** [review]
LGTM, lets see what CI says.

**Comment by janiversen** [comment]
Thanks for the patch, would be nice if you find other things to submit.

**Pull request merged**